src/input/input_json.rs

```diff
@@ -114,6 +114,15 @@ impl Input for JsonInput {
             _ => err_val_error!(input_value = InputValue::InputRef(self), kind = ErrorKind::SetType),
         }
     }
+
+    fn strict_bytes(&self) -> ValResult<Vec<u8>> {
+        match self {
+            JsonInput::String(s) => Ok(s.clone().into_bytes()),
+            JsonInput::Int(int) => Ok(int.to_ne_bytes().to_vec()),
```

Comment:
I stumbled on this issue too while writing this. After giving this a thought, there's really no way to know exactly what value the user wants. I'd imagine arbitrarily choosing coercion will add frustration.

> if there's a (single) obvious representation in the field type AND converting to that looses no information, coerce, otherwise error.

This sounds like a good rule of thumb. If there's no single obvious representation, letting the user work around it will remove ambiguity.

Reply:
Agreed 👍.